### PR TITLE
Fix a crash if a client disconnects during RDM discovery.

### LINKED ISSUE #950: Crash during RDM discovery
I suspect we're closing & deleting the RPC channel and then later the RDM discovery callback is run. 

We should funnel the discovery requests via the PortBroker to avoid this, just like we do with Get / SET rdm requests. 

Received Segmentation fault: 11
0   libolacommon.0.dylib                0x000000010159e8c2 _ZN12_GLOBAL__N_1L17_DumpStackAndExitEi + 194
1   libsystem_platform.dylib            0x00007fff8b7f7f1a _sigtramp + 26
2   ???                                 0x00007fff5ec520b0 0x0 + 140734783365296
3   libolacommon.0.dylib                0x0000000101615b69 _ZN3ola3rpc10RpcChannel15RequestCompleteEPNS0_18OutstandingRequestE + 249
4   libolacommon.0.dylib                0x00000001015adc9f _ZN3ola18SingleUseCallback0IvE3RunEv + 15
5   libolaserver.0.dylib                0x0000000100fbe97f _ZN3ola18SingleUseCallback1IvRKNS_3rdm6UIDSetEE3RunES4_ + 15
6   libolaserverplugininterface.0.dylib 0x000000010146fb99 _ZN3ola8Universe17DiscoveryCompleteEPNS_13BaseCallback1IvRKNS_3rdm6UIDSetEEE + 185
7   libolaserverplugininterface.0.dylib 0x000000010145e29f _ZN3ola18SingleUseCallback0IvE3RunEv + 15
8   libolaserverplugininterface.0.dylib 0x0000000101471580 _ZN3ola13MultiCallback3RunEv + 32
9   libolaserverplugininterface.0.dylib 0x0000000101464aef _ZN3ola18SingleUseCallback1IvRKNS_3rdm6UIDSetEE3RunES4_ + 15
10  libolacommon.0.dylib                0x00000001015f2131 _ZN3ola3rdm33DiscoverableQueueingRDMController17DiscoveryCompleteERKNS0_6UIDSetE + 65
11  libolacommon.0.dylib                0x00000001015f3f2f _ZN3ola18SingleUseCallback1IvRKNS_3rdm6UIDSetEE3RunES4_ + 15
12  libolausbdmx.0.dylib                0x00000001013248c3 _ZN3ola18SingleUseCallback2IvbRKNS_3rdm6UIDSetEE3RunEbS4_ + 19
13  libolacommon.0.dylib                0x00000001015cfbd1 _ZN3ola3rdm14DiscoveryAgent13SendDiscoveryEv + 321
14  libolacommon.0.dylib                0x00000001015cea24 _ZN3ola3rdm14DiscoveryAgent14BranchCompleteEPKhj + 356
15  libolausbdmx.0.dylib                0x0000000101324162 _ZN3ola3usb20JaRulePortHandleImpl11DUBCompleteEPNS_13BaseCallback2IvPKhjEENS0_16USBCommandResultEhhRKNSt3__112basic_stringIhNS8_11char_traitsIhEENS8_9allocatorIhEEEE + 194
16  libolausbdmx.0.dylib                0x000000010132494f _ZN3ola18SingleUseCallback4IvNS_3usb16USBCommandResultEhhRKNSt3__112basic_stringIhNS3_11char_traitsIhEENS3_9allocatorIhEEEEE3RunES2_hhSB_ + 15
17  libolausbdmx.0.dylib                0x000000010132b46f _ZN3ola17MethodCallback2_0INS_3usb16JaRuleWidgetPortENS_18SingleUseCallback0IvEEvPNS_13BaseCallback4IvNS1_16USBCommandResultEhhRKNSt3__112basic_stringIhNS7_11char_traitsIhEENS7_9allocatorIhEEEEEENS2_12CallbackArgsEE5DoRunEv + 367
18  libolausbdmx.0.dylib                0x0000000101316bef _ZN3ola18SingleUseCallback0IvE3RunEv + 15
19  libolacommon.0.dylib                0x00000001015adbd1 _ZN3ola2io12SelectServer15DrainAndExecuteEv + 209
20  libolacommon.0.dylib                0x00000001015b130e _ZN3ola2io12SelectPoller16CheckDescriptorsEP6fd_setS3_ + 446
21  libolacommon.0.dylib                0x00000001015b0a7e _ZN3ola2io12SelectPoller4PollEPNS0_14TimeoutManagerERKNS_12TimeIntervalE + 1006
22  libolacommon.0.dylib                0x00000001015ad13c _ZN3ola2io12SelectServer14CheckForEventsERKNS_12TimeIntervalE + 204
23  libolacommon.0.dylib                0x00000001015ad02b _ZN3ola2io12SelectServer3RunEv + 123
24  olad                                0x0000000100fac4b0 main + 1504
25  libdyld.dylib                       0x00007fff8da7f5c9 start + 1

### PR DESCRIPTION
**Pull request by nomis52**
Closes #950

### CODE REVIEW

File: olad/ClientBroker.cpp [hunk 1]
```diff
@@ -54,6 +54,21 @@ void ClientBroker::SendRDMRequest(const Client *client,
                         callback));
 }
 
+void ClientBroker::RunRDMDiscovery(const Client *client,
+                                   Universe *universe,
+                                   bool full_discovery,
+                                   ola::rdm::RDMDiscoveryCallback *callback) {
+  if (!STLContains(m_clients, client)) {
+    OLA_WARN << "Running RDM discovery but the client doesn't exist in the "
```
Comment (peternewman) [inline olad/ClientBroker.cpp]: Is it worth printing the client's UID to try and help identifying it?

File: olad/ClientBroker.cpp [hunk 2]
```diff
@@ -54,6 +54,21 @@ void ClientBroker::SendRDMRequest(const Client *client,
...
+             << "broker!";
+  }
```
Comment (peternewman) [inline olad/ClientBroker.cpp]: Is there any point running discovery if the client is now AWOL?

**Comment by peternewman** [comment]
Generally LGTM

File: olad/ClientBroker.cpp (hunk 1 above)
Comment (nomis52) [inline olad/ClientBroker.cpp]: This isn't an error. It's just what happens.

File: olad/ClientBroker.cpp (hunk 2 above)
Comment (nomis52) [inline olad/ClientBroker.cpp]: This indicates a bug, the client may be added later so we should continue.

Comment (peternewman) [inline olad/ClientBroker.cpp]: Hence do we not want the client's UID to help fix the bug?

Comment (nomis52) [inline olad/ClientBroker.cpp]: It's not a bug on the client side, so knowing which client it is isn't really going to help.

Comment (peternewman) [inline olad/ClientBroker.cpp]: Ah okay, I assumed it was.

**Pull request merged**